docs/examples/classification.ipynb

```diff
@@ -0,0 +1,946 @@
+    "with pd.option_context('display.max_rows', 8):\n",
+    "    display(model_coefs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.4 Gradient Boosted Decision Trees\n",
+    "We will now train a `HistGradientBoostingClassifier`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "# Note that this cell might take a little while.\n",
+    "from sklearn.preprocessing import OrdinalEncoder\n",
+    "from sklearn.experimental import enable_halving_search_cv\n",
+    "from sklearn.model_selection import HalvingGridSearchCV\n",
+    "from sklearn.ensemble import HistGradientBoostingClassifier\n",
+    "\n",
+    "\n",
+    "\n",
+    "col_trans_bt = ColumnTransformer(\n",
+    "    [\n",
+    "        (\"categorical_features\", OrdinalEncoder(), x_discrete),\n",
+    "    ]\n",
+    ")\n",
+    "m_hgbt_base = Pipeline(\n",
+    "    [\n",
+    "        (\"column_transformer\", col_trans_bt),\n",
+    "        (\n",
+    "            \"model\", HistGradientBoostingClassifier(\n",
+    "                loss=\"log_loss\",\n",
+    "                categorical_features=list(range(len(x_discrete))),\n",
+    "                max_iter=200,\n",
+    "                random_state=33,\n",
+    "            )\n",
+    "        ),\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "param_grid = {\n",
+    "    \"model__learning_rate\": [0.02, 0.05],\n",
+    "    \"model__l2_regularization\": [0.1, 0.2, 0.3, 1],\n",
+    "    \"model__max_depth\": [None, 1, 2, 3, 5],\n",
+    "    \"model__max_leaf_nodes\": [3, 10, 30, 50]\n",
+    "}\n",
+    "\n",
+    "\n",
+    "# successive halfing grid search (CV) on the training data\n",
+    "shgs = HalvingGridSearchCV(\n",
+    "    m_hgbt_base,\n",
+    "    param_grid=param_grid,\n",
+    "    cv=5,\n",
+    "    scoring=\"neg_log_loss\",\n",
+    "    random_state=321,\n",
+    ").fit(X_train, y_train)\n",
+    "\n",
+    "m_hgbt = shgs.best_estimator_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "({'model__l2_regularization': 0.1,\n",
+       "  'model__learning_rate': 0.05,\n",
+       "  'model__max_depth': None,\n",
+       "  'model__max_leaf_nodes': 50},\n",
+       " -0.012953509513355315)"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "shgs.best_params_, shgs.best_score_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "200"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "m_hgbt[-1].n_iter_"
```

Comment:
This results in 200, i.e. the maximum we set. This shows that we need to change some parameters for the HGBT.

Ideally, HGBT can use `early_stopping`. But the default value "auto" uses early stopping only when there are at least 10'000 training samples. Therefore, I would set it manually to True and then play again with some hyperparameters. The idea is that the best ones a not at the boundaries.